vowpalwabbit/learner.h

```diff
@@ -104,23 +121,54 @@ typedef void (*tend_example)(vw& all, void* d, example& ec);
   func_data end_pass_fd;
   func_data end_examples_fd;
   func_data finisher_fd;
+  test_example_data test_example_fd;
 
 public:
+  bool inline multiline_learn() const { return learn_fd.learn_multiline_f != nullptr; }
```

Comment:
We should be able to assume that things are either multiline for both learn & predict or neither learn & predict.